### ISSUE TITLE
🐛 lowercase patch crashes fiber

### ISSUE THREAD

**Issue by dioslibre** [gofiber/fiber#716]
**Fiber version**
1.13.3

**Issue description**
Lowercase patch crashes fiber, 'PATCH' works fine. This problem does not happen with other methods.

**Code snippet**

```go
package main

import "github.com/gofiber/fiber"

func main() {
  app := fiber.New()

  // Steps to reproduce

 app.Post("/api/one/:model", u.VerifyToken, d.Create)
// await fetch(url, { method: 'post', opts }) ==>  OK
// await fetch(url, { method: 'POST', opts }) ==> OK

 app.Patch("/api/one/:model", u.VerifyToken, d.Update)
// await fetch(url, { method: 'patch', opts }) ==>  crash
// await fetch(url, { method: 'PATCH', opts }) ==> OK

  app.Listen(3000)
}
```

**Comment by ReneWerner87** [comment]
thanks for the bug report

after some tests with, curl, httpie and node-fetch and several different constellations we don't know exactly how you produced the bug, please provide a step by step guide to the bug

**Comment by dioslibre** [comment]
First, I should mention that the work you did with fiber is awesome, Thanks a million. 

Why would you test with curl, httpie, node-fetch and all the several different constellations  and not test with browser fetch?! perhaps I should have been more specific.

Anyway, here's a simple working example demonstrating what I'm talking about :
[https://github.com/dioslibre/fiber-patch-bug](url)

**Comment by kiyonlin** [comment]
Hi @dioslibre I ran your demo, and I found `fetch` send an `OPTIONS` request first before `PATCH` in my chrome so that `fiber` returns `405  Method Not Found`.

**Comment by dioslibre** [comment]
hey @kiyonlin, I'm using chrome also, I don't get  `405 Method Not Found`, I get `net::ERR_CONNECTION_REFUSED` and fiber crashes.
![Screenshot_2020-08-14_12-57-55](https://user-images.githubusercontent.com/5884640/90247179-f98ec780-de2d-11ea-833e-9f28335545c1.png)

**Comment by dioslibre** [comment]
@Fenny it also crashes with curl, and not only with lowercase patch, but lowercase post also. I ran this `curl -X post http://localhost:3000/api` and it crashed. What did you do to test this on curl?

**Comment by kiyonlin** [comment]
I already fixed this bug(#672), but it hasn't released yet.  Under normal circumstances, http methods are all uppercase.

**Comment by dioslibre** [comment]
@kiyonlin 'Under normal circumstances, http methods are all uppercase' Yes, you're right.
Thank you for taking the time, and for coding this beautiful library.
I'll close this for now.
